java_api/standard_api/src/main/java/uk/ramp/samples/Samples.java

```diff
@@ -0,0 +1,51 @@
+package uk.ramp.samples;
+
+import static uk.ramp.samples.Sampler.sampleFrom;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.util.List;
+import org.apache.commons.math3.distribution.RealDistribution;
+import org.immutables.value.Value.Immutable;
+import uk.ramp.distribution.Distribution;
+import uk.ramp.parameters.Component;
+
+@Immutable
+@JsonSerialize
+@JsonDeserialize
+public interface Samples extends Component {
+  List<Number> samples();
+
+  @JsonIgnore
+  default Number mean() {
+    double mean = samples().stream()
+        .mapToDouble(Number::doubleValue)
+        .average()
+        .orElseThrow();
+
+    if (mean % 1 == 0) {
+      return (int) mean;
```

Comment:
what's the scenario where we want an int here? This feels like curious behaviour to me

Reply:
Good spot, this is not needed. If the user of this API call needs an int, they will be able to call number.intValue().
Updated PR. Thanks.